### PR TITLE
Remove unneeded fields in NXsource

### LINKED ISSUE #145: Remove unneeded fields in NXsource
In [this commit](https://github.com/FAIRmat-NFDI/nexus_definitions/commit/d94a5cef37f6b59414218aa609f005fa3d3a4578), some fields were re-added to NXsource which are currently not used anymore by the MPES application definitions. We should check carefully where else they are in use (e.g., in [APM](https://fairmat-nfdi.github.io/nexus_definitions/classes/contributed_definitions/NXapm.html)) and then we can decide which fields to delete/keep.

None of these new fields are part of the [NIAC version of NXsource](https://github.com/nexusformat/definitions/blob/main/base_classes/NXsource.nxdl.xml) at the moment.

### PR DESCRIPTION
**Pull request by lukaspie**
Fixes #145.

### CODE REVIEW

**Review by domna — COMMENTED** [review]
Do we want to keep photon_energy, peak_power and pulse_energy?

These were added in https://github.com/FAIRmat-NFDI/nexus_definitions/commit/d94a5cef37f6b59414218aa609f005fa3d3a4578#diff-4e4b88fc8f8c8d6b8bb68c4b78653e28da62550a3b431b39422821e0f7c14a90, too. But are not part of the original NXsource.

**Comment by lukaspie** [comment]
peak_power is used in {NXbeam_path](https://fairmat-nfdi.github.io/nexus_definitions/classes/contributed_definitions/NXbeam_path.html#index-0)

**Comment by lukaspie** [comment]
I built the APM and EM branches locally and could not find any use of photon_energy, so I removed it.

What I did find though is that in [NXapm](ttps://github.com/FAIRmat-NFDI/nexus_definitions/blob/c043182ebf571641c675303349c9471b4a9e2ade/contributed_definitions/nyaml/NXapm.yaml#L459), the field `wavelength` is used what is not part of the NXsource base class. So I propose to add it here instead of `center_wavelength`.

**Comment by domna** [comment]
> I built the APM and EM branches locally and could not find any use of photon_energy, so I removed it.
> 
> What I did find though is that in NXapm, the field `wavelength` is used what is not part of the NXsource base class. So I propose to add it here instead of `center_wavelength`.

Maybe it's better if we discuss this with @mkuehbach as I think we decided for mpes that we want to collect such things in `NXbeam` while keeping `NXsource` only to collect fabrication infos or beamline data (with making it only recommended, too). In this view it would be better that apm also uses wavelengths in NXbeam and not in source.

**Comment by lukaspie** [comment]
I suggest we keep this PR open and just link it in the APM PR for discussion.

**Comment by lukaspie** [comment]
@mkuehbach would you please approve this PR?

File: base_classes/nyaml/NXsource.yaml [hunk 1]
```diff
@@ -134,32 +134,15 @@ NXsource(NXobject):
       type: NX_DATE_TIME
       doc: |
         date and time of the most recent injection.
-  photon_energy(NX_FLOAT):
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply base_classes/nyaml/NXsource.yaml:137]: Indeed I think we dropped the prefix photon at some and left energy which makes sense that an NXsource has many type in its enumeration and thus allowing only for "photon" as the prefix implies is misleading however, indeed photon_energy is a concept that likely not everybody has changed yet and as soon as we hit the NIAC meeting or consolidation with nexusdefinitions we would drop here silently

**Review by mkuehbach — DISMISSED** [review]

File: base_classes/nyaml/NXsource.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukaspie) [reply base_classes/nyaml/NXsource.yaml:137]: No, as I said above, this PR is about things that WE added, but are not in use anymore. This field `photon_energy` was added by us in the context of MPES, but is not in use anymore. You can check [here](https://fairmat-nfdi.github.io/nexus_definitions/search.html?q=photon_energy) that it is not used by any other application definition (also NXstxm does not use it). `NXsource` has a field `energy` that is used all over the place, but we are not touching it.

**Comment by lukaspie** [comment]
In summary, I have removed the following fields from NXsource base class:
- `photon_energy` -> was added in MPES-context, not used anywhere
- `bunch_number_start` -> was added in MPES-context, not used anywhere
- `bunch_number_end` -> was added in MPES-context, not used anywhere
- `center_wavelength` -> renamed to `wavelength` as used in APM

From the fields that were added in https://github.com/FAIRmat-NFDI/nexus_definitions/commit/d94a5cef37f6b59414218aa609f005fa3d3a4578, I have kept the following:
- `pulse_energy`: used in NXapm [here](https://github.com/FAIRmat-NFDI/nexus_definitions/blob/c043182ebf571641c675303349c9471b4a9e2ade/contributed_definitions/nyaml/NXapm.yaml#L522).
- `peak_power`: used in [NXbeam_path](https://fairmat-nfdi.github.io/nexus_definitions/classes/contributed_definitions/NXbeam_path.html#index-12)

So, nothing that is used outside of the FAIRmat context was altered and the fields that are still in use within FAIRmat have not been touched. So the PR should be okay. There seems to be an issue with the nyaml/nxdl consistency check, so this needs to be resolved before merging the PR.

**Comment by mkuehbach** [comment]
Hi @lukaspie from my point of view you can resolve the remaining conversations. Please coordinate with @domna though as with this PR we also have many XML files modified which is not what this PR intends to primarily as it is about NXsource.

For NXsource and context one should know that pulse_energy in atom probe should not be with the source but with a specific atom probe base class NXpulser, I will keep this in mind when implementing the pynxtools-apm and make the relevant change to the appdef if necessary then.

**Comment by lukaspie** [comment]
> Hi @lukaspie from my point of view you can resolve the remaining conversations. Please coordinate with @domna though as with this PR we also have many XML files modified which is not what this PR intends to primarily as it is about NXsource.
> 
> For NXsource and context one should know that pulse_energy in atom probe should not be with the source but with a specific atom probe base class NXpulser, I will keep this in mind when implementing the pynxtools-apm and make the relevant change to the appdef if necessary then.

Changes to NXDL files gone after rebase, now only changes in NXsource.

**Review by sanbrock — APPROVED** [review]
LGTM

**Pull request merged**